tests/test_execute.py

```diff
@@ -34,16 +35,17 @@ def doctree():
     def doctree(
         source, config=None, return_warnings=False, entrypoint="jupyter_sphinx"
     ):
-        src_dir = tempfile.mkdtemp()
-        source_trees.append(src_dir)
-        with open(os.path.join(src_dir, "conf.py"), "w", encoding = "utf8") as f:
-            f.write("extensions = ['%s']" % entrypoint)
-            if config is not None:
-                f.write("\n" + config)
-        with open(os.path.join(src_dir, "contents.rst"), "w", encoding = "utf8") as f:
-            f.write(source)
+        src_dir = Path(tempfile.mkdtemp())
+        source_trees.append(src_dir.as_posix())
```

Comment:
I think this is sufficient:

```suggestion
        source_trees.append(src_dir)
```